### PR TITLE
[backport ipa-4-6] Backport one-way trust to AD with shared secret to ipa-4-6

### PR DESCRIPTION
**Pull request by abbra**
Backport of https://github.com/freeipa/freeipa/pull/2926 to ipa-4-6. Includes additional patch with commit https://github.com/freeipa/freeipa/commit/703497532abe4189835d0a02b32f9919c889bc1c that wasn't backport to ipa-4-6 at the time it was introduced.

### CODE REVIEW

**Comment by abbra** [comment]
@tiran, added two backports to reduce difference between 4.6 and 4.7 -- mostly for paths.* constants used to call utilities around.
Python2 linter on Travis CI produces an error for relative import that I cannot reproduce with python2-pylint on F29. Any help is appreciated.

File: ipaserver/install/plugins/adtrust.py [hunk 1]
```diff
@@ -1,29 +1,26 @@
...
-# (at your option) any later version.
-#
-# This program is distributed in the hope that it will be useful,
-# but WITHOUT ANY WARRANTY; without even the implied warranty of
-# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-# GNU General Public License for more details.
-#
-# You should have received a copy of the GNU General Public License
-# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+# Copyright (C) 2012-2019  FreeIPA Contributors see COPYING for license
```
Comment (tiran) [inline ipaserver/install/plugins/adtrust.py]: Same add, add ``from __future__ import absolute_import`` after the comment.

**Comment by tiran** [comment]
I was able to push from __future__ import absolute_import into your branch.

**Comment by tiran** [comment]
ipa-4-6:

* 9a453619c8c46322fc652f57376e4e834f957f57 Replace hard-coded paths with path constants
* 076d89429d838b472782f20da483841847e435b5 Support Samba 4.9
* b70e4d19d8bb610fedc50c8ad17cb7b61229247f Add design page for one-way trust to AD with shared secret
* 7a7ef33fa8d1e800467e7bda5e23170edbabb4a2 trust: allow trust agents to read POSIX identities of trust
* b2bac94c2486523f180a16068c5c07df8169aa1f trusts: add support for one-way shared secret trust
* 7476953c85eddd92e7a2f010a34a3ba421c91965 upgrade: upgrade existing trust agreements to new layout
* 9ce3a29915fa691369fff8401b5995e0f7f06e06 upgrade: add trust upgrade to actual upgrade code

**Pull request closed without merging**